### PR TITLE
Add div decomposition in ArmQuantizer

### PR DESCRIPTION
**Pull request by Erik-Lundell**
Suggestion or what a pass to decompose a div in ArmQuantizer might look like.

### CODE REVIEW

File: backends/arm/passes/arm_pass_manager.py [hunk 1]
```diff
@@ -45,3 +50,8 @@ def transform_to_backend_pipeline(
                     self.add_pass(AnnotateChannelsLastDimOrder())
 
         return self._transform(graph_module)
+
+    def transform_for_annotation_pipeline(self, graph_module: torch.fx.GraphModule):
+        self.add_pass(DecomposeDivPass())
```
Comment (digantdesai) [inline backends/arm/passes/arm_pass_manager.py:55]: I guess the motivation for [this API](https://github.com/pytorch/pytorch/blob/183c32fd3b665414dc3700c2747da55406b14156/torch/ao/quantization/quantizer/quantizer.py#L138-L150) is clear and this is allowed, however I do want to make sure modifying graph in the Quantizer is something we are OK with from general ET lowering point of view. I also see SDPA as a valid use case.

cc @kimishpatel, @mergennachin - any comments?

Comment (Erik-Lundell) [reply backends/arm/passes/arm_pass_manager.py:55]: Ok - I'll resolve the merge conflict and mark this as ready for review tomorrow unless I get new comments.

**Review by digantdesai — APPROVED** [review]

File: backends/arm/passes/scalars_to_attribute_pass.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    targeted_ops = [
+        torch.ops.aten.add.Tensor,
+        torch.ops.aten.sub.Tensor,
+        torch.ops.aten.sub_.Tensor,
+        torch.ops.aten.mul.Tensor,
+        torch.ops.aten.div.Tensor,
+    ]
+
+    def call(self, graph_module: GraphModule) -> GraphModule:
```
Comment (digantdesai) [inline backends/arm/passes/scalars_to_attribute_pass.py:31]: return type should be PassResult?

Comment (Erik-Lundell) [reply backends/arm/passes/scalars_to_attribute_pass.py]: Nice catch, thanks!

**Comment by Erik-Lundell** [comment]
> still failing with `executorch/backends/arm/operators/op_reciprocal.py:44:46 Incompatible parameter type [6]`

I was not able to find this error in the logs of the failed jobs here on GH and could not reproduce this locally but I rebased and forced-push and now only one job fails from what I can see

File: backends/arm/operators/op_reciprocal.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        tosa_graph: ts.TosaSerializer,
+        inputs: List[TosaArg],
+        output: TosaArg,
+        is_quant_node: bool,
+    ) -> None:
+        # 1/X
+
+        if is_quant_node:
+            input = inputs[0]
+            input_qargs = get_quant_node_args(node.args[0])
```
Comment (digantdesai) [inline backends/arm/operators/op_reciprocal.py:44]: New internal error

executorch/backends/arm/operators/op_reciprocal.py:44:46 Incompatible parameter type [6]: In call `get_quant_node_args`, for 1st positional argument, expected `Node` but got `Union[None, List[typing.Any], Dict[str, typing.Any], bool, complex, float, int, range, slice, str, SymBool, SymFloat, SymInt, device, dtype, layout, memory_format, OpOverload, Tensor, Node, typing.Tuple[typing.Any, ...]]`.

**Comment by digantdesai** [comment]
Also rebase when you fix this 😓

File: backends/arm/operators/op_reciprocal.py (hunk 3 above)
Comment (Erik-Lundell) [reply backends/arm/operators/op_reciprocal.py]: Ok, I assume this is due to some internal extra type checker? I realized that get_quant_node_args was called differently than in other node visitors, hopefully fixing that solves the error.

**Pull request closed without merging**